File: pset6/readability/readability.py
from cs50 import get_string

# Capturar texto
text = get_string("Digite o seu texto: ")

letras = 0
palavras = 1
frases = 0

# Loop para verificar cada caractere do texto
for i in text:
    if i.isalpha():
        letras += 1
    elif i == " ":
        palavras += 1
    elif i == "." or i == "!" or i == "?":
        frases += 1

L = (letras / palavras) * 100
S = (frases / palavras) * 100

# Calculo de Colleman
index = round(0.0588 * L - 0.296 * S - 15.8)

# Resultado
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability/readability.c
#include <cs50.h>
#include <string.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
//Declarando as variaveis
    int letras = 0;
    int palavras = 1;
    int frases = 0;
    float resultado;

//Capturar texto do usuário
    string s = get_string("Insira seu texto: ");

//cria um loop verificando todos caracteres do texto
    for (int i = 0, n = strlen(s); i < n; i++)
    {
        //verifica se o caractere é alguma letra maiuscula ou minuscula
        if ((s[i] >= 'a' && s[i] <= 'z') || (s[i] >= 'A' && s[i] <= 'Z'))
        {
            letras++;
        }
        //verifica se o caractere é um espaço (o que significa que acabou uma palavra)
        else if (s[i] == ' ')
        {
            palavras++;
        }
        //verifica se o caractere é algum tipo de pontuação, para contar as frases
        else if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            frases++;
        }
    }

//Calcula a quantidade de letras e frases para cada 100 palavras
    float letrasm = ((float) letras / palavras) * 100;
    float frasesm = ((float) frases / palavras) * 100;

//Coleman
    resultado = ((0.0588 * letrasm) - (0.296 * frasesm) - 15.8);

    int a = (int) round(resultado);

    if (a < 0)
    {
        printf("Before Grade 1\n");
    }
    else if (a > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.i\n", a);
    }
}